homeassistant/components/abode/sensor.py

```diff
@@ -34,8 +34,12 @@
 
     devices = []
     for device in data.abode.get_devices(generic_type=CONST.TYPE_SENSOR):
-        for sensor_type in SENSOR_TYPES:
-            devices.append(AbodeSensor(data, device, sensor_type))
+        if CONST.TEMP_STATUS_KEY in device.get_value(CONST.STATUSES_KEY):
```

Comment:
```py
entities = []
for sensor_type in SENSOR_TYPES:
    if sensor_type not in device.get_value(CONST.STATUSES_KEY):
        continue
    entities.append(AbodeSensor(data, device, sensor_type))
```

Reply:
Much cleaner, thanks!

Only addition I made was changing `async_add_entities(devices)` just below to `async_add_entities(entities)` since the name of the list changed to `entities`.